### ISSUE TITLE
High pitched sounds out of tune

### ISSUE THREAD

**Issue by backwardspy** [kitao/pyxel#150]
First off, thanks for all your hard work on Pyxel. It is a fantastic package!

**Describe the bug**
When using the music editor, I have noticed that the high pitched notes do not seem to be in tune. For example, if I play each octave of D, the lower ones sound correct but the highest is out of tune.

**To Reproduce**
Open the music editor and play the same note at different octaves including the highest.

**Expected behavior**
All instances of the same note should sound in tune.

**Desktop (please complete the following information):**
 - OS: macOS 10.14.5 and Windows 10

If you have any suggestions as to what might be causing this, I am happy to play around with the source code and submit a pull request if I manage to fix the problem.

Thanks again!

**Comment by kitao** [comment]
Pitch is calculated by Channel::NoteToPitch method

```cpp
inline int32_t Channel::NoteToPitch(float note) {
  return 440.0f * pow(2.0f, (note - 33.0f) / 12.0f);
}
```

Though I have not clearly identified the cause, the accuracy of this function is a problem?

**Comment by backwardspy** [comment]
If I get the time, I will spend some time debugging that function to see if I can spot any problems with it.

**Comment by kitao** [comment]
I fixed this in 1.2.5.
It's caused by the period of sound wave treated as integer. I changed it as float.
